internal/repository/sync.go

```diff
@@ -0,0 +1,95 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/policy"
+	"github.com/gittuf/gittuf/internal/rsl"
+)
+
+var (
+	ErrCloningRepository = errors.New("unable to clone repository")
+	ErrDirExists         = errors.New("directory exists")
+	ErrPushingRepository = errors.New("unable to push repository")
+	ErrPullingRepository = errors.New("unable to pull from remote")
+)
+
+// Clone wraps a typical git clone invocation, fetching gittuf refs in addition
+// to the standard refs. It performs a verification of the RSL against the
+// specified HEAD after cloning the repository.
+// TODO: resolve how root keys are trusted / bootstrapped.
+func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Repository, error) {
+	if dir == "" {
+		// FIXME: my understanding is backslashes are not used in URLs but I haven't dived into the RFCs to check yet
+		split := strings.Split(strings.TrimSpace(strings.ReplaceAll(remoteURL, "\\", "/")), "/")
```

Comment:
I'll look into this! I wasn't sure about treating this as a URL as it can also be a file path (that's also how we test right now).

Reply:
What you might want to do then is feed EscapedPath into filepath.Split, which should also handle the \'s from Windows

Reply:
I took a pass at using this by pulling it into a helper. I'm not sure if we benefit because (unless I'm doing something wrong), EscapedPath gives me nothing for windows file paths. I'm not yet using filepath.Split because we need the EscapedPath to work on Windows first.